ion-schema/src/constraint.rs

```diff
@@ -2042,9 +2042,11 @@ impl ConstraintValidator for ValidValuesConstraint {
                         },
                         ValidValue::Element(element) => {
                             // get value without annotations
-                            let value = value.value();
+                            let value: IonData<_> = value.value().into();
+                            let actual_value: IonData<_> = element.value().into();
 
-                            if element.value().ion_eq(value) {
+                            // this comparison uses the Ion equivalence based on Ion specification
+                            if actual_value == value {
```

Comment:
FYI—It should also be possible to do this:

```rust
if IonData::eq(value.value(), element.value()) {
```

Reply:
It seems to require types for both the arguments of `IonData::eq()`.